src/projectController.ts

```diff
@@ -279,8 +275,8 @@ export class ProjectController implements vscode.Disposable {
             const config: ConfigurationReader | undefined = workspaceContext.config;
             if (config) {
                 this.sourceDirectorySub.set(folder, config.onChange('sourceDirectory', async (sourceDirectories: string | string[]) => this.doSourceDirectoryChange(folder, sourceDirectories)));
-                this.buildDirectorySub.set(folder, config.onChange('buildDirectory', async () => this.refreshDriverSettings(folder, config.sourceDirectory)));
-                this.installPrefixSub.set(folder, config.onChange('installPrefix', async () => this.refreshDriverSettings(folder, config.sourceDirectory)));
+                this.buildDirectorySub.set(folder, config.onChange('buildDirectory', async (buildDirectory: string) => this.refreshDriverSettings(folder, config.sourceDirectory, buildDirectory)));
+                this.installPrefixSub.set(folder, config.onChange('installPrefix', async () => this.refreshDriverSettings(folder, config.sourceDirectory, config.buildDirectory((projects && projects?.length > 1) || false))));
```

Comment:
MultiProject state is determined by the length of the sourceDirectory setting, not the projects array. But you can just delete this when the function is updated to not require this parameter.